src/main/scala/com/gu/formstack/services/FormstackService.scala

```diff
@@ -0,0 +1,161 @@
+package com.gu.formstack.services
+
+import cats.Monoid
+import cats.implicits._
+import com.gu.formstack.config.{ Config, FormstackJsonConfiguration }
+import com.gu.formstack.services.FormstackClient.{ DeleteEntityResponse, Form, Forms, Submission, Submissions }
+import io.circe.generic.extras.ConfiguredJsonCodec
+
+import java.time.ZonedDateTime
+import scala.util.Try
+
+class FormstackService(
+  client: FormstackClient,
+  isDryRun: Boolean // i.e. if true, only HTTP requests with no side-effects will be executed.
+) {
+
+  import FormstackService._
+
+  private def checkNotDryRun: Try[Unit] =
+    Either.cond(!isDryRun, (), DryRun).toTry
+
+  def deleteForm(form: Form): Try[DeletionResult] = {
+    (for {
+      _ <- checkNotDryRun
+      response <- client.deleteForm(form)
+    } yield {
+      DeletionResult.nonDryRun(response)
+    }).recover {
+      case DryRun => DeletionResult.dryRun(DeleteEntityResponse(form.id, success = 0))
+    }
+  }
+
+  def deleteSubmission(submission: Submission): Try[DeletionResult] = {
+    (for {
+      _ <- checkNotDryRun
+      response <- client.deleteSubmission(submission)
+    } yield {
+      DeletionResult.nonDryRun(response)
+    }).recover {
+      case DryRun => DeletionResult.dryRun(DeleteEntityResponse(submission.id, success = 0))
+    }
+  }
+
+  // Paginate through the pages of forms,
+  // combining the results of applying a function to each page.
+  def forEachPageOfForms[A: Monoid](f: Forms => Try[A]): Try[A] = {
```

Comment:
Open to making this less generic and having a function of type `Forms => Try[Unit]` instead...